### PR TITLE
Healthcheck bug fixes 

### PR DESCRIPTION
**Pull request by baburgess**

### CODE REVIEW

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: packages/arb-node-core/staker/inboxReader.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (ir *InboxReader) getMessages(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	if ir.healthChan != nil {
+	if ir.healthChan != nil && from != nil {
```
[the remark this answers is not in the document]
Comment (baburgess) [reply packages/arb-node-core/staker/inboxReader.go:85]: There was a case were we hit a seg fault due to trying to use Set with a nil value in InboxReader. I added the checks just to make sure we avoided that if they were ever set to nil.

**Review by joshuacolvin0 — APPROVED** [review]
approving merge

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**